cognite_toolkit/_cdf_tk/storageio/_selectors.py

```diff
@@ -26,3 +28,19 @@ def as_filter(self) -> dict[str, Any]:
             data_set_external_ids=[self.data_set_external_id] if self.data_set_external_id else None,
             asset_subtree_external_ids=[self.hierarchy] if self.hierarchy else None,
         )
+
+
+@dataclass(frozen=True)
+class InstanceSelector: ...
+
+
+@dataclass(frozen=True)
+class InstanceFileSelector(InstanceSelector):
+    datafile: Path
```

Comment:
I have included this just to demonstrate why I want to have `InstanceSelector` as a base class, i.e., there are multiple ways of selecting instances. See https://github.com/cognitedata/toolkit/pull/1874 for the implementation of it.